front_end/panels/ai_chat/ui/settings/providers/GroqSettings.ts

```
@@ -0,0 +1,234 @@
+
+        // Get all Groq models including any custom ones
+        const allGroqModels = this.getModelOptions('groq');
+        const actualModelCount = groqModels.length;
+
+        // Get current mini and nano models from storage
+        const miniModel = getStorageItem(MINI_MODEL_STORAGE_KEY, '');
+        const nanoModel = getStorageItem(NANO_MODEL_STORAGE_KEY, '');
+
+        // Refresh existing model selectors with new options if they exist
+        if (this.miniModelSelector) {
+          refreshModelSelectOptions(this.miniModelSelector as any, allGroqModels, miniModel, i18nString('defaultMiniOption'));
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)